program/freedom-metal-program.wake

```diff
@@ -1,4 +1,28 @@
 # FIXME
+def addMetalMemory type memSubnode inDTSContents =
+  def subnodeOpt =
+    match "({memSubnode}@[0-9]+)".stringToRegExp
+      Pass regex =
+        def nodeRegex = regExpCat (`^\t\t`, `(?:L[0-9]+:\s)?`, regex, `\s{$`, Nil)
```

Comment:
I just want to point out that this is making the assumption that all DTS labels are of the form `L[0-9]+`, which is only coincidentally true for the DTS files generated by rocket-chip. If we/they/someone ever decides to change this, it'll break this. You might want to write a more generic regex for matching any kind of label. According to the [spec](https://github.com/devicetree-org/devicetree-specification/releases/download/v0.2/devicetree-specification-v0.2.pdf), the valid characters in a label are `[a-zA-Z0-9_]`.